external/mdal/frmts/mdal_ply.hpp

```diff
@@ -28,31 +28,23 @@ namespace MDAL
       DriverPly *create() override;
 
       bool canReadMesh( const std::string &uri ) override;
+      int faceVerticesMaximumCount() const override {return 100;}
```

Comment:
100 faces per vertices? I try to create a mesh with this format, and it seems to support only triangles as faces when saving it.

Reply:
I am not sure where and why you are seeing that - the unit tests include writing quads - although not larger faces. The format places no constraint - so any constraint is probably an unwanted side-effect somewhere.

Reply:
Under QGIS, I tried to create a new mesh from an existing one with quads and with PLY format for the new one. In the result I had only triangles as the implementation of PLY format does not support quads... Quite weird results... I tried to reproduce with a mesh of the tests of QGIS, this time,  it worked fine with quads... 

I have to investigate more to understand...

Reply:
ok, I got it but bad first diagnostic. here what I had:

**original mesh**

![original](https://user-images.githubusercontent.com/7416892/135956831-f721bb72-0e0f-4086-a287-ba92b9bb3a94.png)

**PLY mesh**

![ply](https://user-images.githubusercontent.com/7416892/135956850-777e548b-da51-4034-b0fb-cba2b3cd80d1.png)

So I believed that was unsupported quads, but I was completely wrong (there was also quads)... sorry for that.

The problem comes from the coordinate in the PLY file:

![image](https://user-images.githubusercontent.com/7416892/135956988-6f987df0-f7f4-408c-8b52-bac4cde748a4.png)


There are a method to avoid that in MDAL (used for 2dm format):
`std::string coordinateToString( double coordinate, int precision = 2 );`
https://github.com/lutraconsulting/MDAL/blob/f799685648e48ff82d13f3299287b91ba5db1b4f/mdal/mdal_utils.hpp#L101
`std::string doubleToString( double value, int precision = 6 );`
https://github.com/lutraconsulting/MDAL/blob/f799685648e48ff82d13f3299287b91ba5db1b4f/mdal/mdal_utils.hpp#L105

Reply:
https://github.com/lutraconsulting/MDAL/issues/379